File: Examples/CSharp/ModifyingAndConvertingImages/PNG/SpecifyTransparency.cs
// GIST-ID: 45a2de98ee07b655953f1415a250ff44
using Aspose.Imaging.FileFormats.Png;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API references 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.PNG
{
    class SpecifyTransparency
    {
        public static void Run()
        {
            Console.WriteLine("Running example SpecifyTransparency");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_PNG();

            // Initialize variables to hold width and height values.
            int width = 0;
            int height = 0;

            // Initialize an array of type Color to hold the pixel data.
            Color[] pixels = null;

            // Create an instance of RasterImage and load a BMP image.
            using (RasterImage raster = (RasterImage)Image.Load(dataDir + "aspose_logo.png"))
            {
                // Store the width and height in variables for later use.
                width = raster.Width;
                height = raster.Height;

                // Load the pixels of RasterImage into the array of type Color.
                pixels = raster.LoadPixels(new Rectangle(0, 0, width, height));
            }

            // Create and initialize an instance of PngImage while specifying size and PngColorType.
            using (PngImage png = new PngImage(width, height, PngColorType.TruecolorWithAlpha))
            {
                // Save the previously loaded pixels onto the new PngImage and set the TransparentColor property to specify which color should be rendered as transparent.
                png.SavePixels(new Rectangle(0, 0, width, height), pixels);
                png.TransparentColor = Color.Black;
                png.HasTransparentColor = true;
                png.Save(dataDir + "SpecifyTransparencyforPNGImages_out.jpg");
            }

            Console.WriteLine("Finished example SpecifyTransparency");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/PNG/SpecifyTransparency.vb
Imports Aspose.Imaging.FileFormats.Png

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.PNG
    Class SpecifyTransparency
        Public Shared Sub Run()
            ' ExStart:SpecifyTransparency
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_PNG()

            'Initialize variables to hold width & height values
            Dim width As Integer = 0
            Dim height As Integer = 0

            'Initialize an array of type Color to hold the pixel data
            Dim pixels As Color() = Nothing

            ' Create an instance of RasterImage and load a BMP image
            Using raster As RasterImage = DirectCast(Image.Load(dataDir & Convert.ToString("aspose_logo.png")), RasterImage)

                ' Store the width & height in variables for later use
                width = raster.Width
                height = raster.Height

                ' Load the pixels of RasterImage into the array of type Color
                pixels = raster.LoadPixels(New Rectangle(0, 0, width, height))
            End Using

            ' Create & initialize an instance of PngImage while specifying size and PngColorType
            Using png As New PngImage(width, height, PngColorType.TruecolorWithAlpha)
                ' Save the previously loaded pixels on to the new PngImage
                png.SavePixels(New Rectangle(0, 0, width, height), pixels)

                ' Set TransparentColor property to specify which color to be rendered as transparent
                png.TransparentColor = Color.Black
                png.HasTransparentColor = True

                ' Save the result on disc
                png.Save(dataDir & Convert.ToString("SpecifyTransparencyforPNGImages_out.jpg"))
            End Using
            ' ExEnd:SpecifyTransparency
        End Sub
    End Class
End Namespace